### PR TITLE
Solidify button appearance combos, document active state

### PR DESCRIPTION
**Pull request by kylesuss**
* Solidify the button `appearance`s -- you can no longer combine multiple string appearances.
* Remove unnecessary `css` calls from within the `styled` blocks
* Pull common string constants out to a variable so that they can easily be changed in the future
* Add PropTypes for the `active` state. I do not really understand the point of this state though and I would like to add a comment for it for docs mode, so I would like to understand the use case
* Add stories for `active` state

--

Over time, I would like to slim up the styles in this component as I think they are a bit hard to follow. It can be hard to tell where certain styles are coming from given there are some duplications in various prop checks and properties. Also, given the styles take up ~300 or so lines, I think there is room to split this out in a way that is a bit easier to track down. It feels like an unnecessary optimization now though, so I believe this would be better served for future work/once it actually becomes a problem.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@storybook/design-system",
-  "version": "0.0.10",
+  "version": "0.0.11",
```
Comment (kylesuss) [inline package.json]: I would like to pull this into the learn storybook repo, so I think it needs another release

File: src/components/Button.js [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import React, { Fragment } from 'react';
 import PropTypes from 'prop-types';
-import styled, { css } from 'styled-components';
+import styled from 'styled-components';
```
Comment (kylesuss) [inline src/components/Button.js]: As mentioned, these `css` calls are not necessary in this context

File: src/components/Button.js [hunk 3]
```diff
@@ -441,16 +464,17 @@ Button.propTypes = {
    Buttons with icons by themselves have a circular shape
   */
   containsIcon: PropTypes.bool,
-  size: PropTypes.oneOf(['small', 'medium']), // this is enum incase we need to add more sizes
+  size: PropTypes.oneOf(Object.values(SIZES)),
```
Comment (kylesuss) [inline src/components/Button.js]: Thoughts on this approach? The array of values is not immediately apparent, though I like that using the variable means that there is only 1 place where the string value needs to be updated. In docs mode, the array of values looks great.

File: src/components/Button.js [hunk 4]
```diff
@@ -233,6 +251,7 @@ const ButtonWrapper = styled.button`
           &:focus {
             box-shadow: ${color.secondary} 0 0 0 3em inset,
               ${rgba(color.secondary, 0.4)} 0 1px 9px 2px;
+            color: ${color.lightest};
```
Comment (kylesuss) [inline src/components/Button.js]: To prevent this in the focus state:

Screen Shot 2019-06-01 at 2 59 21 PM

Now it looks like this:
Screen Shot 2019-06-01 at 2 59 10 PM

File: src/test.js [hunk 5]
```diff
@@ -1,7 +1,7 @@
...
+import ExampleComponent from './index';
 
 describe('ExampleComponent', () => {
   it('is truthy', () => {
-    expect(ExampleComponent).toBeTruthy()
-  })
-})
+    expect(ExampleComponent).toBeTruthy();
+  });
+});
```
Comment (kylesuss) [inline src/test.js]: Prettier updates..

File: src/components/Button.js [hunk 6]
```diff
@@ -421,6 +443,7 @@ export function Button({
 }
 
 Button.propTypes = {
+  active: PropTypes.bool,
```
Comment (kylesuss) [inline src/components/Button.js]: I tend to prefer using qualifiers for booleans, like `isActive` or `isUnclickable` because I think the cognitive load is lower. It makes it obvious that the value will be true or false without having to look up the docs. That said, I think it should be agreed upon as a standard so I didn't update anything -- just calling it out to get the conversation started.

**Comment by kylesuss** [comment]
@domyen Button updates! Left you a bunch of comments here to get some conversations started. Can you help me out w/ the purpose of the `active` button state as well? I'd like to get some documentation for it in the PropTypes to be clear about the use case.

File: src/components/Button.js (hunk 6 above)
Comment (domyen) [reply src/components/Button.js]: That approach sgtm!

File: src/components/Button.js (hunk 3 above)
Comment (domyen) [reply src/components/Button.js]: 👍

File: src/components/Button.js [hunk 7]
```diff
@@ -364,64 +297,34 @@ const ButtonWrapper = styled.button`
...
   if (isLink) {
     return (
-      <ButtonLink
-        loading={loading}
-        appearance={appearance}
-        size={size}
-        containsIcon={containsIcon}
-        {...props}
-      >
+      <ButtonLink isLoading={isLoading} disabled={isDisabled} {...props}>
```
Comment (kylesuss) [inline src/components/Button.js]: `disabled` is always a tricky one w/ regard to the `is` prefixing -- since the default HTML property is just `disabled`, I typically pass that to the actual button element so that it can do things naturally, but require the prefix on the React component that wraps it.

**Comment by kylesuss** [comment]
:rocket: PR was released in v0.0.13 :rocket:
